File: apps/api-dotnet/src/Planora.Application/Features/Analysis/Dtos/StartAnalysis/AnalysisOptionsDto.cs
namespace Planora.Application.Features.Analysis.Dtos.StartAnalysis;

public sealed record AnalysisOptionsDto(
    bool IncludeTopography = true,
    bool IncludeSoil = true,
    bool IncludeBearing = true,
    bool IncludeRisk = true,
    bool IncludeBorehole = true,
    decimal? ContourInterval = null,
    List<decimal>? SlopeCategories = null,
    string? ReferencePlane = null,
    List<string>? SoilDepths = null);

File: apps/frontend/src/app/features/analyses/interfaces/start-analysis/analysis-options-dto.ts
export interface AnalysisOptionsDto {
  includeTopography?: boolean;
  includeSoil?: boolean;
  includeBearing?: boolean;
  includeRisk?: boolean;
  includeBorehole?: boolean;
  contourInterval?: number;
  slopeCategories?: number[];
  referencePlane?: string;
  soilDepths?: string[];
}